### DISCUSSION TITLE
Lore: Hyperspace Weaponry & Implications

### DISCUSSION THREAD

**Discussion by Zitchas** [endless-sky/endless-sky#7061] [Lore]
Given that we're getting into a lot of detail regarding hyperspace and what we can do with it, I think it might be beneficial to pull this out of the PR discussion and into a stand-alone discussion thread that will be easier to deal with in the future.

----

quoted from me:
> Could you elaborate (as in write out in full) what your objections are to hyperspace utilization in weaponry? I'm having trouble grasping your opposition to this. Basically everything humanity has ever figured has been used directly or indirectly as a way of killing or harming others, so having something that is flat out "impossible to weaponize by anyone, anywhere, anywhen, anyhow" just seems... illogical? Immersion breaking? I'm not saying that the description as it stands is perfect (and I know Ravenshining is planning to revise it), but as a general principle I don't like the idea of labelling this (or any) particular thing as utterly impossible to ever use as part of a weapon of some kind.
>
> More to the point, we're dealing with a sci-fi story with an immense range of technologies and capabilities, and it just doesn't make sense to me that this would be a hard immutable law of the universe that is constant across every faction, every technology tree, every kind of understanding of the universe.

----

All of the following is from @Amazinite, posted in the Korath Exiles Expansion, Revised - Stage I PR. I haven't put it in indentation because that'd just make it hard to read. I'll post any of my own words in a new post. Original post by Amazinite is here: https://github.com/endless-sky/endless-sky/pull/6891#issuecomment-1198720276

---
All stories need constants. I don't think that the fact that we're working in the sci-fi genre means that any and all barriers should be capable of being broken. There must be some things which are true no matter the faction or perspective in question, otherwise you have no grounding and essentially anything can happen. While that can lead to cool and spectacular looking moments, it's not healthy for an overall story, particularly when what is done for the sake of being cool or creating some sense of awe directly contradicts what we've seen elsewhere, especially so when no explanation is given.

As Arachi mentioned, the hyperspace ramming from that one Star Wars movie is the perfect example of something being done for the sake of being cool but which contradicts everything known about hyperdrives/hyperspace in the Star Wars universe and leaves the story worse off because of it. Although a distinction to be made might be that an instance such as that doesn't just contradict everything known by a specific faction within the universe, but by all factions and peoples anywhere within the universe at any time. It's perfectly fine to contradict the ideas of a single faction. This is already done when the Pug arrive in the Free Worlds campaign and cut the hyperspace links, something which humanity didn't think possible. It's where something contradicts everything seen everywhere that problems arise.

The Heaver, as you yourself mentioned, is somehow throwing objects through hyperspace without the objects themselves being hyperspace capable. As hyperspace is understood from the entire rest of the game, the hyperdrive itself must be part of what travels through hyperspace; there is no "hyperspace window" that the hyperdrive creates and it's only incidental that the hyperdrive goes through hyperspace as well because it's part of the ship traveling. So from what we can tell, something about the hyperdrive itself moving is integral to accessing hyperspace, an aspect that this weapon lacks. (Notably, the term "unstable hyperspace window" comes from the original PR from Nomadic, and the term "hyperspace window" was also used in #4952 in part of Nomadic's contribution to that PR. No where else in the game is this idea of a "hyperspace window" mentioned.)

Even throwing out the "hyperspace window" term and making use of "hyperlink" as Raven has suggested, which at least is an established term within the game (admittedly used very sparsely, but at least it's a term widely recognized by players), doesn't solve the problems with this weapon contradicting established lore. From what we an tell from hyperlinks from the end of the Free Worlds campaign as well as from the lore dump document from MZ, hyperlinks can only be formed by having some device at two separate points in space, and something about those devices is what creates the "gravitational slipstream" between them to allow access to hyperspace, slipstreams that we only ever see made across vast distances of space. By contrast, using the hyperlink idea, this weapon would somehow be capable of creating such a link across a short distance from only one point in space.

Speaking of, this seems like a good time to actually look at the news blurb that was mentioned:
> "Many a hyperspace-based weapon has been tested, but due to the dynamics of entering hyperspace while not aligned with a hyperspace lane, all of them have failed. Thankfully."

This blurb explicitly calls out the use of a hyperspace-based weapon that tries to work while misaligned with proper hyperlinks, which is exactly what the Heaver is currently. While we don't know exactly what the "dynamics of entering hyperspace while not aligned with a hyperspace lane" entail, I'd say that this blurb implies that humanity has at least attempted to make a device not too dissimilar to the Heaver as it is described currently. What then would allow the Korath to succeed where humanity has failed?

Now I'm not saying that just because something has never been seen doesn't mean it can never be done. Perhaps there's nothing about the in-universe mechanics of hyperspace that contradict the ability to send stuff into hyperspace without that stuff having a hyperdrive attached to it, or for someone to make a short-ranged link while only having control of one end of the link. But when you introduce new mechanisms to a system as ubiquitous as hyperspace travel, those mechanisms need to be explained, even if only at a low level, and such an explanation should include why we haven't seen these mechanisms anywhere else. That is, everyone has access to hyperspace, so why have only the Korath managed to create such technology, at least from among the factions we've seen, which include much older and/or more powerful factions?

But should we keep the Heaver as a strictly gravity-based weapon, I don't think we need to answer any questions of "Why don't we see this elsewhere," because I think we do. We explicitly see this from the Heliarchs and their Attractors and Repulsors, weaponry that one would presume works similarly to the Banisher/Expeller, and if the Banisher and Expeller are said to be gravity based, then so too would the Attractor and Repulsor. We also see this from the repulsor engines that every human ship are said to have, which, while not weapons, show that manipulation of gravity is possible. Similarly, I think one could argue that we implicitly see this in the Quarg with their ringworlds, structures that are created using materials mined from planets and moons within the system, an act that would be made much quicker if the Quarg had the ability to move worlds, so I don't think it would contradict anything to say that the Quarg are capable of that. And there are plans to add tractor beams as human tech for pulling flotsams in, which would in all likelihood involve some sort of gravitational manipulation.

So the manipulation of gravity is already well established as a mechanic within the universe of Endless Sky, something that even humanity is capable of on a small scale. It doesn't take much imagination then to say "The Korath can do that but better." I think it does take far more to say "The Korath can do these things with hyperspace that no one else can."

<hr>

I suppose a TL;DR would be that while I don't imagine it would be impossible to explain the Heaver as a hyperspace weapon, I think we'd need to be very careful with how we explain it in order to not contradict what we already see. The main two points among my wall of text were the use of hyperspace without a hyperdrive and the use of hyperspace without using the hyperlinks, so let's see if we can brainstorm something to resolve these two.

On hyperdrive-less hyperspace travel:
* As I described, all hyperspace travel at the moment involves the hyperdrive being part of the travel. If hyperspace travel is possible without having a hyperdrive attached to yourself, that would imply that warp gates are possible. But if warp gates are possible, why do we see everyone using hyperdrives?
* The explanation from Nomadic on the mechanics of this weapon was that it creates a "hyperspace window." As I pointed out, this term isn't used anywhere in the game. Therefore, if we want to introduce this term, I think we need to define what it is and how it might work.
* Perhaps warp gate technology is possible, but is more difficult to develop than hyperdrives, which as we can see are actually rather small, being only 20 tons.
* Perhaps a normal hyperdrive does work by bringing itself through hyperspace, but it is possible to open a "hyperspace window" to allow anything through without it needing a hyperdrive, but such windows would require a large amount of energy and fuel to keep open long enough to allow something the size of a ship through, a further reason for why hyperdrives would be more prevalent in the galaxy than warp gates.
* The Korath then could perhaps be capable of creating small, unstable hyperspace windows open for just long enough to send their langrage in before it collapses shortly afterward, unleashing the langrage into real space at considerable speeds.
* The collapsing of the hyperspace window is what brings the matter back into real space; by contrast, perhaps a hyperdrive can not be aborted once activated due to its self-contained nature, hence why normal hyperdrives can't be weaponized by humanity.

The above says nothing on hyperlinks though, which seem to be the structure that actually allows for travel within hyperspace; using the warp gate idea, two warp gates would perhaps still only be able to open a hyperspace window along an existing hyperlink.

On hyperlinks:
* As Raven said earlier, there is a link between hyperspace and gravitational manipulation, as the hyperlinks are described as "gravitational slipstreams." One would imagine that such links are created by manipulating gravity between two points.
* Let's say the Korath are adept at manipulating gravity, moreso than any of the other factions in the game that are below T3.
* Perhaps the Korath are both capable of opening hyperspace windows as well as creating temporary, very short-ranged hyperlinks, links short enough that one end of the link could be at the ship and the other end could be created just beyond the ship, meaning that the Korath still have control over both ends of a link in order to create it.

Therefore, the lore behind the Heaver becomes the following:
* Hyperspace windows are comparatively less efficient means of traveling through hyperspace, but allow an object to travel without having a hyperdrive attached to it, and the collapsing of a hyperspace window allows the traveling matter to be weaponized in a way that something attached to a hyperdrive can't.
* Hyperspace windows still require a hyperlink in order to be used, but the Korath are capable enough in manipulating gravity to form an incredibly short-ranged and short-lived hyperlink between two points near a ship.
* The Heaver then is capable of creating these short links, opening a hyperspace window at either end, and rapidly chucking material into it while collapsing the window shortly afterward (i.e. before the material reaches the other end of the window, which would presumably bring it to a near stop just as ships do when jumping) in order to drag the material back into real space at high speeds, allowing it to damage ships.
* If anything is too close to the Heaver when it tries creating a link, the link collapses and releases the energy that otherwise would have gone into accelerating the langrage into a forceful push, hence the first stage of the Heaver's projectile not dealing damage. (The first stage could perhaps be changed to an explosion so that the firing ship also gets knocked about by the collapsing link.)

If the Heaver is a "hyperspace-based weapon" based off of these ideas then I can see it working out.

**Comment by Zitchas** [comment]
First off, thanks for the in-depth reply and analysis. I appreciate it, and will respond to it, but will need some time to digest and compose a reply. :)

**Comment by MasterOfGrey** [comment]
Cross-posted:

Per usual, when Amazinite takes the time to express themselves fully, their logic is pretty damn good where lore and technology intersect.

There's a *subsequent* issue though from that conclusion.

> Hyperspace windows still require a hyperlink in order to be used, but the Korath are capable enough in manipulating gravity to form an incredibly short-ranged and short-lived hyperlink between two points near a ship.

This is where the problem starts. If they're gravity-based slipstreams, that presumably work well between systems *because systems are big, concentrated sources of gravity that you can leverage*, then producing a small hyperlink to use would imply a greater degree of technical finesse than linking two systems together would - because you lack the leverage of an entire system to anchor it.

The expected consequence of this is that if they could create a temporary link to use as a weapon, they'd have already surpassed the understanding necessary to build their own hyperlinks - an understanding that would naturally have been pre-collapse; and would have seen a Korath invasion of other territories to act as a pressure-valve release, instead of their self-contained implosion.

Accordingly; I am now *against* the idea of a hyperspace weapon.

**Reply by Amazinite** [reply]
> This is where the problem starts. If they're gravity-based slipstreams, that presumably work well between systems *because systems are big, concentrated sources of gravity that you can leverage*, then producing a small hyperlink to use would imply a greater degree of technical finesse than linking two systems together would - because you lack the leverage of an entire system to anchor it.

I did think about whether this implies that the Korath can make their own hyperspace links while writing up that response, and I definitely agree that the answer should be no for the sake of the story elsewhere, so resulting from that I think we do need to provide an explanation that says that creating links on a small scale like this is easier than doing the same on a large scale.

While entire star systems are greater sources of gravity, they're also much, much further from one another. By comparison, at the size scale of a ship/a single weapon the Korath have much less gravity to leverage, but they're also creating a much shorter link. Keep in mind too that time is a factor; the links between systems are seemingly permanent, while the mini-links that the Heaver are creating last shorter than a second. So given that these weaponized links are extremely small and extremely short-lived, I think it can be said that they're easier to create than the extremely long, permanent hyperlinks that we see between systems.

**Reply by MasterOfGrey** [reply]
So here's the thing right, that logic implies that making hyperspace/hyperlinks work as a weapon is really just a matter of... well maths.

And frankly, maths is the easy part. Where maths is concerned we're usually somewhere between several decades to several centuries ahead of our actual technological progress for implementing that maths. - And at least a few someone's understand the hyperspace maths well enough to build the drive after all. 

At best, I would expect that the less gravity that's involved, the more power it would take to create, otherwise I don't see why most civilisations wouldn't develop hyperspace weapons *first* and the drives *later*. (Especially since, presumably, keeping a whole ship intact and stable through a long jump would be harder than just dropping a tungsten slug through a split-second moment of time.)

**Reply by Amazinite** [reply]
I think that's where the importance of the "hyperspace window" idea comes in; there has to be something exotic about opening windows into hyperspace that makes it more difficult to do compared to using a hyperdrive. It's the use of hyperspace windows and creating short links combined that leads to the weaponry, not creating short links alone. Whatever that "something exotic" is can easily be hand-waved, though, just as with the specifics of how a hyperdrive actually puts you into hyperspace.

Important to note too that one doesn't need to understand how to create links of any size in order to understand how to create a hyperdrive to use them, another reason why drives could come first and weapons come (much) later.

**Reply by MasterOfGrey** [reply]
That does raise a question too actually, about whether you actually *enter* hyperspace. Visually in-game, it certainly doesn't appear like you do any such thing. With the way things are now it feels more like you just "surf" on regular space using some kind of hyperplane that's built into this reality - which also harks back to the differences between the warp and weft in the lore document.

**Reply by Amazinite** [reply]
Another important note: humanity took 150 years from developing the hyperdrive to developing repulsor engines, so taking advantage of existing hyperlinks is seemingly much easier than doing any form of gravity manipulation, at least from our sample size of one.

**Comment by Zitchas** [comment]
Three thoughts, none of which constitute a response:

1) The weapon description refers to temporary hyperlinks. This might be problematic for the reasons mentioned.
2) That being said, the Korath have reason for not creating hyperlinks because that's not advantageous for them. They can *already* bypass areas with no links to raid or attack or whatever. Creating hyperspace links would just enable their enemies to follow them back. So strong motivation to *not* do that. Perhaps not even expend much energy developing it, either.
3) We are dealing with a race that has learned how to hyperjump without using links at all. Instead of focusing on creating temporary links, perhaps they have some way to impart momentum to a bubble or something to make it "fire" where it should go.

**Reply by MasterOfGrey** [reply]
Regarding 2: we know that building jump drives is a hazardous industrial process - we can also infer from the automata and from the way jump drives are distributed among Exile ships that a) hyperdrives could be distributed like candy, and b) just because a ship is important doesn't mean it has a jump drive, those are given to ships that *use* them.
Also, your enemy being able to capture a meaningful number of jump drives and use them against you is a non-trivial concern.

I don't think you could reasonably assert that the Korath at their prime, when they were feeling the squeeze, would have either been threatened by or felt any moral compunctions about connecting to either Hai or Human space and launching an invasion far in excess of anything we have thus far conceived of in in-game terms.

**Comment by Amazinite** [comment]
I think something important to define would be what it actually means to weaponize hyperspace.

Weaponizing hyperspace is not simply using hyperspace as a means of transportation for a weapon. Ships are equipped with weapons, and ships can travel through hyperspace, but that isn't *exactly* "weaponizing hyperspace." Saying that would be like saying that militaries weaponize roads when they use them to transport tanks.

To me, weaponizing hyperspace means using hyperspace as a means to enhance your weapon's capabilities, particularly by accelerating your weapon/its projectiles such that it has a bigger impact.

**Comment by AwesomeNoob999** [comment]
Maybe the weapons need jump drive type tech, as the korath can make jd's and probs have learned enough from that to make a micro "Hyperspace" field, it would also explain why most factions don't have warp gates, as it would be way too expensive for the korath and the quarg might have them in their ringworld

**Reply by AwesomeNoob999** [reply]
Most other factions don't know how jd's work, and in the case of the pug, they can make wormholes, which are probs cheaper to sustain

**Comment by VaelophisNyx** [comment]
My two cents would just be to ask: Why can't it just be a basic magnetic acceleration weapon. All you'd need even for simple garbage is a thin slab of metal, like shotgun wadding, and a sufficiently powerful magnet and you're set. No dancing around hyperspace lore needed

**Comment by xobes** [comment]
Re:
> So the manipulation of gravity is already well established as a mechanic within the universe of Endless Sky, ...

Perhaps I haven't played enough to know and I haven't scoured all the outfits to find out: if once can manipulate gravity, can there be cargo/outfit expansions that grant more room than they take?

**Reply by Saugia** [reply]
No; the original dev MZ, coming from EV, wanted to make sure expansions were properly balanced and had their own downsides. One of the things he wanted to avoid from EV was expansions that gave tons without much impact in return, so every ton given to the player/a ship in general needs to have some sort of offset.

**Reply by Saugia** [reply]
Perhaps, in terms of "alien magic" it is possible, maybe the Quarg or Pug can do this, but for the game's balance it would be a problem to introduce something obtainable that gave a ship space that wasn't all accounted for. (Maybe as a one-of item? Unsure.)

**Reply by esksp** [reply]
> if once can manipulate gravity, can there be cargo/outfit expansions that grant more room than they take?

You'd need to not only change gravity, but also _space_: Reducing mass (weight) won't change the volume required.
You would need to manipulate space-time, i.e. create some sort of "cargo pocket dimension", and in this case you've opened an infinitely big can of worms, story-wise... :-D
